Madness/String.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  String.swift
+//  Madness
+//
+//  Created by Josh Vera on 10/19/15.
+//  Copyright © 2015 Rob Rix. All rights reserved.
+//
+
+import Foundation
+
+public typealias CharacterParser = Parser<String.CharacterView, Character>.Function
+public typealias StringParser = Parser<String.CharacterView, String>.Function
+public typealias DoubleParser = Parser<String.CharacterView, Double>.Function
+
+public func digit() -> CharacterParser {
+	return oneOf("123456789")
+}
+
+public func double() -> DoubleParser {
+	let minus = (%"-")|?
+	let digits = digit()+
+	
+	return minus >>- { sign in
+		digits >>- { digits in
+			let double = digits.reduce(0.0, combine: { num, char in
+				return 10 * num + Double(String(char))!
+			})
+
+			return pure(sign != nil ? double : -1 * double)
+		}
+	}
+}
+
+public func oneOf(input: String) -> CharacterParser {
+	return satisfy { input.characters.contains($0) }
+}
+
+public func noneOf(input: String) -> CharacterParser {
+	return satisfy { !input.characters.contains($0) }
+}
+
+public func space() -> CharacterParser {
+	return satisfy { $0 == " " }
+}
+
+public func newline() -> CharacterParser {
+	return satisfy { $0 == "\n" }
+}
+
+public func crlf() -> CharacterParser {
+	return satisfy { $0 == "\r" }
+}
+
+public func endOfLine() -> CharacterParser {
+	return newline() <|> crlf()
+}
+
+public func tab() -> CharacterParser {
+	return satisfy { $0 == "\t" }
+}
```

Comment:
From an API point of view, I wonder if these would be nicer as `let`s instead of `func`s, e.g.:

```swift
public let tab: CharacterParser = satisfy { $0 == "\t" }
```